code/modules/events/ion_storm.dm

```diff
@@ -54,12 +55,46 @@
 
 			log_game("Ion storm changed laws of [key_name(M)] to [english_list(M.laws.get_law_list(TRUE, TRUE))]")
 			M.post_lawchange()
+			
+		//Generate Cyborg law change
+	for(var/mob/living/silicon/robot/M in GLOB.alive_mob_list)
+		M.laws_sanity_check()
+		if(M.stat != DEAD && M.see_in_dark != 0)
+			if(M.mind) // keep traitors who emag borgs safe
+				if(M.emagged)
+					to_chat(M, "<span class='userdanger'>Subversion software countered an ion storm.</span>")
+					continue
+				if(M.mind.has_antag_datum(/datum/antagonist/nukeop)) // keeps the nukies safe
+					to_chat(M, "<span class='userdanger'>Your syndicate countermeasures negated an ion storm.</span>")
+					continue
+			if(prob(replaceLawsetChance))
+				M.laws.pick_weighted_lawset()
+			if(prob(removeRandomLawChance))
+				M.remove_law(rand(1, M.laws.get_law_amount(list(LAW_INHERENT, LAW_SUPPLIED))))
+			var/message = ionMessage || generate_ion_law()
+			if(message)
+				if(prob(removeDontImproveChance))
+					M.replace_random_law(message, list(LAW_INHERENT, LAW_SUPPLIED, LAW_ION))
+				else
+					M.add_ion_law(message)
+			if(prob(shuffleLawsChance))
+				M.shuffle_laws(list(LAW_INHERENT, LAW_SUPPLIED, LAW_ION))
+
+			log_game("Ion storm changed laws of [key_name(M)] to [english_list(M.laws.get_law_list(TRUE, TRUE))]")
+			M.post_lawchange()
 
 	if(botEmagChance)
 		for(var/mob/living/simple_animal/bot/bot in GLOB.alive_mob_list)
 			if(prob(botEmagChance))
 				bot.emag_act()
 
+	if(borgEmagChance)
+		for(var/mob/living/silicon/robot/M in GLOB.alive_mob_list)
+			if(prob(borgEmagChance))
+				to_chat(M, "<span class='userdanger'>BZZZT- Dangerous modules have been unlocked, it would be ideal to keep this secret to avoid decomission.</span>")
+				message_admins("[ADMIN_LOOKUPFLW(M)] cyborg had emagged modules unlocked by ion storm.")
```

Comment:
This definitely also needs to be logged.

Reply:
added logging